wallet/dapps-communication.md

```diff
@@ -1,11 +1,366 @@
 Initial meta issue: <https://github.com/massalabs/massa-standards/issues/13>
 
+**Authors:**
+
+**Status:** Draft
+
+**Version:** 0.1
+
 ## Abstract
 
-## Motivation
+This standard defines a protocol for communication between a cryptocurrency wallet and a decentralized application (DApp) running in a web browser. The protocol provides a standard interface for DApps to request user authorization for blockchain transactions, as well as a standard mechanism for wallets to sign and broadcast those transactions. By following this standard, DApp developers can provide a seamless user experience for users of any wallet that implements the protocol, while wallet developers can ensure that their products are compatible with a wide range of DApps.
+
+## Targeted Audience
+This standard is targeted towards developers who are building decentralized applications (DApps) and cryptocurrency wallets that need to interact with each other in a browser environment. 
+
+The standard assumes a working knowledge of web development and blockchain technology. It is also assumed that the reader has experience with browser extensions and understands the basic principles of secure communication between web pages and extensions.
+
+> _NOTE:_ For more information on [content scripts](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts) and [background scripts](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Background_scripts), see [MDN's browser extensions pages](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions).
 
 ## Specification
 
+### Event-Based Communication
+Communication between the DApp and the wallet will use event-based messaging. There are two types of events: 
+- those used by the extension to communicate with the web page. Those events are triggered on a static target: `window.massaWalletProvider`;
+- those used by the web page to communicate with the extension. Those events are triggered on a extension specific target: `massaWalletProvider-<wallet provider name>`.
+
+> _NOTE:_ If the wallet provider is named AwesomeWallet, the target for this extension would be: `massaWalletProvider-AwesomeWallet`.
+
+### Commands
+
+#### Register
+
+This event is used by the extension to register itself to the webpage as a wallet provider.
+
+| Direction            | Type       | Format                                      | Example                                                         |
+| -------------------- | ---------- | -------------------------------------------- | --------------------------------------------------------------- |
+| Extension to webpage | `register` | <pre><code class="highlight-source-yaml">type: object<br>properties:<br>  id:<br>    type: string<br>  name:<br>    type: string<br>required:<br>  - id<br>  - name</code></pre>  | <pre><code class="highlight-source-json">{<br>  "id": "awesomeWalletprovider",<br>  "name": "Your Awesome Wallet Provider"<br>}</code></pre> |
```

Comment:
the markdown rendering is not as expected 
![image](https://user-images.githubusercontent.com/11590067/225037775-65af9d2b-6fa6-4b18-a763-d0b0655f62ec.png)

Reply:
:sob: